File: API_JeeWork2021/Models/Wework/TopicModel.cs
using DocumentFormat.OpenXml.Office.CoverPageProps;
using JeeWork_Core2021.Classes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace JeeWork_Core2021.Models
{
    public class TopicModel
    {
        public long id_row { get; set; }
        public string title { get; set; }
        public string description { get; set; }
        public long id_project_team { get; set; }
        public bool email { get; set; }
        public List<TopicUserModel> Users { get; set; }
        public List<FileUploadModel> Attachments { get; set; }
        
    }
    public class TopicUserModel
    {
        public long id_row { get; set; }
        public long id_topic { get; set; }
        public long id_user { get; set; }
        public bool favourite { get; set; }

    }
}

File: BackEnd/BE_jeework/src/app/pages/JeeWork_Core/discussions/Model/Topic.model.ts
import { FileUploadModel } from '../../projects-team/Model/department-and-project.model';
import {BaseModel} from './../../../../_metronic/jeework_old/core/_base/crud/models/_base.model';

export class TopicModel extends BaseModel {
    id_row: number;
    title: string;
    description: string;
    id_project_team: string;
    email: string;
    Users: Array<TopicUserModel> = [];
    Follower: Array<TopicUserModel> = [];
    Attachments: Array<FileUploadModel> = [];

    clear() {
        this.id_row = 0;
        this.title = '';
        this.description = '';
        this.id_project_team = '';
        this.email = '';
    }
}

export class TopicUserModel extends BaseModel {
    id_row: number;
    id_topic: number;
    id_user: number;
    favourite: boolean;
    id_nv: number;

    clear() {
        this.id_row = 0;
        this.id_topic = 0;
        this.id_user = 0;
        this.favourite = false;
    }
}
